core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -52,7 +55,36 @@ private[spark] class TaskDescription(
     val addedFiles: Map[String, Long],
     val addedJars: Map[String, Long],
     val properties: Properties,
-    val serializedTask: ByteBuffer) {
+    private var serializedTask_ : ByteBuffer,
+    private var task_ : Task[_] = null) extends  Logging {
```

Comment:
@kayousterhout I like that idea, but I took a brief look at how much would have to change, and its a ton of changes, since now `TaskSchedulerImpl.resourceOffers` would be returning `TaskDescription`s and `Task`s, which then leads to small changes in a lot more places.   I'm torn, but inclined to agree with @witgo